api/src/YoutubeClient.cpp

```diff
@@ -90,6 +94,21 @@ rapidjson::Document VideosResult::GetSchema() {
                     }
                   },
                   "required": ["viewCount"]
+                },
+                "contentDetails: {
+                  "type": "object",
+                  "properties": {
+                    "contentRating": {
+                      "type": "object",
+                      "properties: {
+                        "ytRating": {
+                          "type": "string",
+                          "pattern": "^[0-9]+$"
```

Comment:
the valid value `ytAgeRestricted` will not match this pattern

Reply:
Sorry, wasn't paying attention. Can I just match exactly on `ytAgeRestricted` or just swap to `a-zA-Z`

Reply:
if the property is omitted when the video isn't age restricted using `^ytAgeRestricted$` and checking for the presence of the property with... iirc `KeyExists` would work. if it's an empty string when it's not age restricted it's probably ok to just omit the pattern then if they add other values in the future it won't break the schema validation and the string comparison in the nsfw check function will work...

Reply:
Should I use `HasMember` here? Something like this..
```
  if (data_.HasMember("ytRating") { return json::StringRef(data_["contentDetails"]["contentRating"]["ytRating"]) == "ytAgeRestricted"; } else { return false; }
```